src/test/ui/const-generics/issue-61336.rs

```diff
@@ -0,0 +1,13 @@
+#![feature(const_generics)]
+//~^ WARN the feature `const_generics` is incomplete and may cause the compiler to crash
+
+fn f<T: Copy, const N: usize>(x: T) -> [T; N] {
+    [x; N]
```

Comment:
Could we also have a test where we make sure that this does the right thing? E.g. pass in `N` explicitly and check the return value.

Reply:
I've added another test. It will error later in the compiler with a "array lengths can't depend on generic parameters" error, emitted when creating the HAIR. From a quick look, it appears that the MIR/HAIR data structures all contain a `u64` for the count, so it isn't trivial to change one or two lines and make this work, probably best left for a follow-up (which I'd be interested in trying to tackle). It's an improvement over an ICE in any case.

Reply:
Ah, okay. Yes, this is certainly better than an ICE, and this gives us a good minimal example to fix.